src/lib/RuleResultRecombinator.ts

```diff
@@ -106,8 +106,10 @@ ${v.message.trim()}
 			column,
 			url
 		} = violation;
+		// The silliness of the triple backslashes in the .replace() calls is because we need to escape the backslash so
+		// it ends up in the output, AND the single/double quotes so the code compiles.
 		return `<testcase name="${fileName}">
-<failure message="${fileName}: ${line} ${message.trim()}" type="${severity}">
+<failure message="${fileName}: ${line} ${message.trim().replace(/'/g, '\\\'').replace(/"/g, '\\"')}" type="${severity}">
```

Comment:
The following node module is already included transitively https://github.com/WebReflection/html-escaper

It would be best to use the library since there are other characters that can cause issues.